lib/node_modules/@stdlib/blas/ext/base/dnannsumkbn2/lib/ndarray.native.js

```diff
@@ -40,27 +41,24 @@ var addon = require( './dnannsumkbn2.native.js' );
 *
 * @example
 * var Float64Array = require( '@stdlib/array/float64' );
-* var floor = require( '@stdlib/math/base/special/floor' );
 *
 * var x = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN, NaN ] );
 * var out = new Float64Array( 2 );
 *
-* var N = floor( x.length / 2 );
+* var N = 5;
 *
 * var v = dnannsumkbn2( N, x, 2, 1, out, 1, 0 );
 * // returns <Float64Array>[ 5.0, 4 ]
 */
 function dnannsumkbn2( N, x, strideX, offsetX, out, strideOut, offsetOut ) {
 	var viewOut;
 	var viewX;
-	if ( strideX < 0 ) {
-		offsetX += (N-1) * strideX;
-	}
-	if ( strideOut < 0 ) {
-		offsetOut += strideOut;
-	}
-	viewX = new Float64Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offsetX), x.length-offsetX ); // eslint-disable-line max-len
-	viewOut = new Float64Array( out.buffer, out.byteOffset+(out.BYTES_PER_ELEMENT*offsetOut), out.length-offsetOut ); // eslint-disable-line max-len
+
+	offsetX = minViewBufferIndex( N, strideX, offsetX );
+	offsetOut = minViewBufferIndex( N, strideOut, offsetOut );
```

Comment:
The error you are getting stems from this line. I'll leave it to you to figure out why. Hint: are `x` and `out` the same length?

Reply:
I suggest studying L62 and L63 from the previous implementation. And I encourage you to not simply copy and paste but reason about the implementation.